examples/daal/cpp_sycl/source/datasource/row_merged_to_sycl_homogen.cpp

```diff
@@ -0,0 +1,147 @@
+/* file: datastructures_usm.cpp */
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+/*
+!  Content:
+!    Example of the use of Unified Shared Memory in Row Merged Numeric Table
+!
+!******************************************************************************/
+
+/**
+ * <a name="DAAL-EXAMPLE-CPP-ROWMERGEDTOSYCLHOMOGEN"></a>
+ * \example datastructures_usm.cpp
+ */
+
+#include "daal_sycl.h"
+#include "service_sycl.h"
+
+using namespace daal;
+using namespace daal::services;
+using namespace daal::data_management;
+
+using daal::data_management::internal::convertToSyclHomogen;
+using daal::data_management::internal::SyclHomogenNumericTable;
+using daal::services::internal::SyclExecutionContext;
+
+uint32_t generateMinStd(uint32_t x) {
+  constexpr uint32_t a = 16807;
+  constexpr uint32_t c = 0;
+  constexpr uint32_t m = 2147483647;
+  return (a * x + c) % m;
+}
+
+/* Compute correlation matrix */
+NumericTablePtr computeCorrelationMatrix(const NumericTablePtr &table) {
+  using namespace daal::algorithms;
+
+  covariance::Batch<> covAlg;
+  covAlg.input.set(covariance::data, table);
+  covAlg.parameter.outputMatrixType = covariance::correlationMatrix;
+  covAlg.compute();
+
+  return covAlg.getResult()->get(covariance::correlation);
+}
```

Comment:
Do we really need to compute correlation matrix in this example?

Reply:
The goal is to check that resulting table can be accepted by an algorithm